core/src/consensus/tendermint/mod.rs

```diff
@@ -254,6 +282,8 @@ impl Tendermint {
         self.view.store(0, AtomicOrdering::SeqCst);
         *self.lock_change.write() = None;
         *self.proposal.write() = None;
+        *self.proposal_parent.write() = Default::default();
```

Comment:
This lock is unlocked when the statement is finished.
Are we okay if the other thread looks the partially updated states?

Reply:
I don't know well. There are too many variables which could be changed by other threads.